validphys2/src/validphys/pineparser.py

```diff
@@ -159,9 +159,16 @@ def pineappl_reader(fkspec):
     is_polarized = pine_rep.key_values().get("polarized") == "True"
 
     # Is it hadronic? (at the moment only hadronic and DIS are considered)
-    hadronic = pine_rep.key_values()["initial_state_1"] == pine_rep.key_values()["initial_state_2"]
+    try:
+        parton1 = pine_rep.key_values()["convolution_particle_1"]
+        parton2 = pine_rep.key_values()["convolution_particle_2"]
+    except KeyError:
+        parton1 = pine_rep.key_values()["initial_state_1"]
+        parton2 = pine_rep.key_values()["initial_state_2"]
+    hadronic = parton1 == parton2
+
```

Comment:
This should probably changed in the future to use the `Grid::convolutions` method to make it future proof; starting with the new file format the convolutions will no longer be stored in metadata.

Reply:
That'd be great, especially since the metadata might not always be as reliable.